config/routes.rb

```diff
@@ -0,0 +1,31 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  root 'works#homepage'
+
+  resources :votes, only: [:index, :show, :create]
+
+  resources :works do
+    resources :votes, only: [:index, :create, :update]
+    member do
+      post 'upvote'
+    end
+  end
+
+  resources :works do
+
+   end
+
+
+  resources :users do
+    resources :votes, only: [:index, :create]
+  end
```

Comment:
Overall in this file, you're playing around and experimenting with how to make the votes paths. Ultimately, I'd say: we don't want any routes paths that are on their own (not-nested), and we don't need the "create" path for a vote nested under a user. (There's also no requirement that would be better off if we nested the votes index route under a user, too, but I could see someone argue for it.)